corehq/ex-submodules/casexml/apps/phone/tasks.py

```diff
@@ -22,3 +28,27 @@ def force_update_cleanliness_flags():
     # that there are no bugs in the weekly task.
     # If we haven't seen any issues by the end of 2015 (so 6 runs) we should remove this.
     set_cleanliness_flags_for_all_domains(force_full=True)
+
+
+@task(queue=ASYNC_RESTORE_QUEUE)
+def get_async_restore_payload(restore_config):
+    """Process an async restore
+    """
+    response = restore_config.generate_payload(async_task=current_task)
+
+    # delete the task id from the task, since the payload can now be fetched from the cache
+    restore_config.cache.delete(restore_config.async_cache_key)
+
+    return response
+
+
+@after_task_publish.connect
+def update_celery_state(sender=None, body=None, **kwargs):
+    """Used to update celery state so we know if an async restore made it to celery
```

Comment:
I'm guessing it's hard to test this right? Either way I think you could put more of the detail in the doc_string.